### ISSUE TITLE
POT support gradient backpropogation or not?

### ISSUE THREAD

**Issue by smiles724** [PythonOT/POT#329]
Hi, thanks for providing such an amazing toolkit. However, I wonder whether it can be used for machine learning projects. I saw a previous issue last year reporting their concerns. At that time, POT failed to do that. Today, I tried a simple trial to exam it. However, the result of tensors is still different from results using numpy arrays. Therefore, I do not dare to adopt POT for my machine learning coding. 

Can you give me more details about it?

![image](https://user-images.githubusercontent.com/39299863/147727407-70cc1f8c-85cd-4f2e-a52b-4a5233e459b1.png)

**Comment by rflamary** [comment]
Hello, when using the torch backend the OT loss can be backpropagated but you should use the ot.sinkhorn2 function that return the loss and can be backpropagated.

We are already using it in machine learning application. 

See some examples below
https://pythonot.github.io/auto_examples/index.html#pot-backend-examples

I'm closing the issue, note that this kind of questions are better asked in the discussion section 
https://github.com/PythonOT/POT/discussions

**Comment by smiles724** [comment]
Thanks for your immediate reply. I already found out the universal application of POT in machine learning. However, as you mentioned, can I get loss using other API functions? For instance, I add the generic regularization using ot.optim.gcg. But it does not return loss? How can I obtain loss in that case? 

Or perhaps the current version only allows us to get the loss of the simplest entropy regularization? Thanks.

**Comment by rflamary** [comment]
when the function return the OT plan (such as gcg) you can compute the loss with `torch.sum(plan*M)` and it will be differentiable

**Comment by smiles724** [comment]
Thanks, I followed the same pattern you use to calculate the loss.
